mantidproject/mantid
# Engg GUI beta/demo fixes and improvements

Fixes a handful of GUI issues identified in a demo/early beta testing (see issue for details). This should close all the issues that were deemed as doable and low risk during the beta fixes period.

**To test:**
- Check code changes:
  -  The changes in EnggUtils.py fix the issue of the spurious new lines in the calibration (.prm) files.
  - A couple of methods have been added to the view to handle the "very visible message when the RB number is empty/invalid" and the status messages.
  - The presenter takes care of updating the status when starting/finishing any significant work (calibrate, focus, rebin, fit). The relevant test cases have been updated to cover this as far as possible.
  - About half of the lines change where format changes. It seems that MSVC applies a slightly different clang formatting to multi-line stream in/out than my emacs package, both accepted by clang-format.
- Unit tests have been added/extended
- Documentation has been updated.
- Functional tests:
  - See what happens when you leave the RB Number field empty
  - Status messages: check that when you start, and then see finishing a calibration, focusing, rebinning, fitting, the status text at the bottom is updated correctly. When fitting peaks fails it should say that no peaks could be fitted.
  - There's a new table workspace: engggui_calibration_banks_parameters, as described in the documentation changes. See that it contains reasonable values and is updated with new calibrations.
  - Output directories: together with the other points, check that the output files are generated both in the `User/RBNumber` directory and in the general/all directory (directly under `EnginX_Mantid`). This includes the '.prm' files (calibration) and for focusing the '.nxs' file and, when enabling the "output files" option the '.dat', '.gss', and '.his' files. Log messages are also generated that say where the files are copied.
  - In the fitting tab, preview plot, check that the left and right tails of the peak shapes are plotted reasonably well. The startX/endX window around the peak centers has been increased to `X0-9*S`, `X0+12*S` when evaluating the peak. Hopefully this is enough.
  - Try to raise any remaining issues in the GUI.

Fixes #16252.

I think this does not deserve any specific release note, as it is just about polishing the interface, fixing recent issues, and a bit of eye candy.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
